### PR TITLE
Allow choice of optimizer and loss

### LINKED ISSUE #254: Add optimizer choice
Choose all optimizers in `torch.optim` with parameters `lr` and `weight_decay` to allow to use them with the commandline.
Right now, Adam is hard-coded.

### PR DESCRIPTION
**Pull request by 14thibea**
To be merged after #241 

Currently the optimizer is hard-coded to `Adam` and the loss depends on the task performed.
It will be now possible to give other choices to the user.

### CODE REVIEW

**Review by mdiazmel — APPROVED** [review]
LGTM!

Can be merged!
